machine/xen/v1alpha1.go

```diff
@@ -0,0 +1,536 @@
+	if err != nil {
+		return nil, fmt.Errorf("could not create xen context: %v", err)
+	}
+	defer xenCtx.Close()
+
+	machine.CreationTimestamp = metav1.Now()
+
+	domID, err := xenCtx.DomainCreateNew(cfg)
+	if err != nil {
+		return machine, fmt.Errorf("could not create xen domain: %v", err)
+	}
+
+	machine.Status.PlatformConfig = domID
+
+	machine.Status.State = machinev1alpha1.MachineStateCreated
+
+	return machine, nil
+}
+func (service *machineV1alpha1Service) Start(ctx context.Context, machine *machinev1alpha1.Machine) (*machinev1alpha1.Machine, error) {
+	if machine.Status.PlatformConfig == nil {
+		return machine, fmt.Errorf("machine has no platform config")
+	}
+
+	domId, ok := machine.Status.PlatformConfig.(xenlight.Domid)
+	if !ok {
+		return machine, fmt.Errorf("machine has no platform config")
+	}
+
+	xenCtx, err := xenlight.NewContext()
+	if err != nil {
+		return nil, fmt.Errorf("could not create xen context: %v", err)
+	}
+	defer xenCtx.Close()
+
+	err = xenCtx.DomainUnpause(domId)
+	if err != nil {
+		return machine, fmt.Errorf("could not unpause xen domain: %v", err)
+	}
+
+	machine.Status.State = machinev1alpha1.MachineStateRunning
+	machine.Status.StartedAt = time.Now()
+
+	return machine, nil
+}
+func (service *machineV1alpha1Service) Pause(ctx context.Context, machine *machinev1alpha1.Machine) (*machinev1alpha1.Machine, error) {
+	if machine.Status.PlatformConfig == nil {
+		return machine, fmt.Errorf("machine has no platform config")
+	}
+
+	domId, ok := machine.Status.PlatformConfig.(xenlight.Domid)
+	if !ok {
+		return machine, fmt.Errorf("machine has no platform config")
+	}
+
+	xenCtx, err := xenlight.NewContext()
+	if err != nil {
+		return nil, fmt.Errorf("could not create xen context: %v", err)
+	}
+	defer xenCtx.Close()
+
+	if err := xenCtx.DomainUnpause(domId); err != nil {
+		return machine, fmt.Errorf("could not unpause xen domain: %v", err)
+	}
+
+	machine.Status.State = machinev1alpha1.MachineStatePaused
+
+	return machine, nil
+}
+func (service *machineV1alpha1Service) Stop(ctx context.Context, machine *machinev1alpha1.Machine) (*machinev1alpha1.Machine, error) {
+	if machine.Status.State == machinev1alpha1.MachineStateExited {
+		return machine, nil
+	}
+
+	domId, ok := machine.Status.PlatformConfig.(xenlight.Domid)
+	if !ok {
+		return machine, fmt.Errorf("machine has no platform config")
+	}
+
+	xenCtx, err := xenlight.NewContext()
+	if err != nil {
+		return nil, fmt.Errorf("could not create xen context: %v", err)
+	}
+	defer xenCtx.Close()
+	if err := xenCtx.DomainDestroy(domId); err != nil {
+		return machine, fmt.Errorf("could not destroy xen domain: %v", err)
+	}
+
+	machine.Status.State = machinev1alpha1.MachineStateExited
+	machine.Status.ExitedAt = time.Now()
+
+	return machine, nil
+}
+func (service *machineV1alpha1Service) Update(ctx context.Context, machine *machinev1alpha1.Machine) (*machinev1alpha1.Machine, error) {
+	panic("not implemented: kraftkit.sh/machine/xen.machineV1alpha1Service.Update")
```

Comment:
Is this how it's done in the other `v1alpha1.go` files?

Reply:
Yes